protocol/protos/NetRemoteWifi.proto

```diff
@@ -3,37 +3,79 @@ syntax = "proto3";
 
 package Microsoft.Net.Remote.Wifi;
 
+import "google/protobuf/field_mask.proto";
 import "WifiCore.proto";
 
-message WifiConfigureAccessPointRequest
+//
+// General notes:
+//
+// Despite all fields being implicitly optional in proto3, all fields are
+// required to be specified unless their definition is explicitly marked with
+// the 'optional' proto keyword.
+//
+
+enum WifiAccessPointApplyConfigurationStatus
+{
+    WifiAccessPointApplyConfigurationStatusUnknown = 0;
+    WifiAccessPointApplyConfigurationStatusSucceeded = 1;
+    WifiAccessPointApplyConfigurationStatusAccessPointInvalid = 2;
+    WifiAccessPointApplyConfigurationStatusAccessPointNotEnabled = 3;
+    WifiAccessPointApplyConfigurationStatusBandInactive = 4;
+    WifiAccessPointApplyConfigurationStatusOperationNotSupported = 5;
+}
+
+message WifiAccessPointApplyConfigurationRequestConfigurationItem
+{
+    // An optional unique identifier for this configuration request.
+    optional string Id = 1;
+
+    // The band the AP should be configured for.
+    Microsoft.Net.Wifi.RadioBand Band = 2;
+
+    // Specifies the parts of the 'Configuration' field to be applied.
+    google.protobuf.FieldMask ConfigurationMask = 3;
```

Comment:
Thinking about this more, this might not be needed at all. In proto3, all fields are implicitly `optional`, so, the client only has to populate the fields they want to configure, and the server can just do a `item.has_<field>()` check to determine if it's present. If it is, then that field/property gets configured. The result structure would still need a `FieldMask` to indicate which fields failed configuration.

I don't know if this moves the needle any on this design or makes or more attractive, but I'm going to make this update for at least the sake of posterity.